### PR TITLE
Allow caller to specify step for DataFilter range

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Allows caller to specify step for a property in Data by:
1. specifying in `properties` on Data
2. specifying in `range` on a given DataFilter.

#### What testing has been done on this PR?
Tested in a storybook example. Added jest test.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [x] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`. NOTE: Used fireEvent to be able to do drag event.
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Related to #6730 

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
Yes. Added `step` to Data `properties.range` and DataFilter `range`.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/DataFilter/__tests__/DataFilter-test.tsx [hunk 1]
```diff
@@ -201,6 +203,54 @@ describe('DataFilter', () => {
...
+        </Data>
+      </Grommet>,
+    );
+
+    const lowerBound = screen.getByRole('slider', { name: 'Lower Bounds' });
+    fireEvent.mouseDown(lowerBound);
+    fireEvent.mouseMove(lowerBound, { clientX: 31, clientY: 20 });
+    fireEvent.mouseUp(lowerBound);
+    expect(lowerBound.getAttribute('aria-valuenow')).toEqual('45');
+    expect(asFragment()).toMatchSnapshot();
```
Comment (jcfilben) [inline src/js/components/DataFilter/__tests__/DataFilter-test.tsx:225]: Do we need a snapshot test here or can we just rely on the check above `expect(lowerBound.getAttribute('aria-valuenow')).toEqual('45');`?

**Review by jcfilben — COMMENTED** [review]
Code looks good
In addition to the `index.d.ts` files should the `propTypes.js` files also be updated?

File: src/js/components/DataFilter/__tests__/DataFilter-test.tsx (hunk 1 above)
Comment (taysea) [reply src/js/components/DataFilter/__tests__/DataFilter-test.tsx:225]: The snapshot is just capturing styling info, while the previous expect is more so testing fuctionailty.

Given there are other snapshots capturing range, should I remove this one to keep things more lean?

Comment (jcfilben) [reply src/js/components/DataFilter/__tests__/DataFilter-test.tsx:225]: Yeah let's remove the snapshot. This article has some good points about snapshot testing: https://kentcdodds.com/blog/effective-snapshot-testing. My main takeaway was that snapshot testing can be very useful, but I think in Grommet we have a tendency to overuse it a bit when the test could be written using other assertions.

Comment (taysea) [reply src/js/components/DataFilter/__tests__/DataFilter-test.tsx:225]: Totally agree, I'll remove.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Comment by taysea** [comment]
Converting to draft because some of the alphabetically sorted filters appear to be missorted in chromatic.

File: src/js/components/DataFilter/DataFilter.js [hunk 2]
```diff
@@ -116,7 +128,13 @@ export const DataFilter = ({
           label
           min={range[0]}
           max={range[1]}
-          step={(range[1] - range[0]) / 20}
+          step={
+            // from `range` on DataFilter
+            rangeProp?.step ||
+            // from range in Data `properties`
+            properties?.[property]?.range?.step ||
+            (range[1] - range[0]) / 20
```
Comment (halocline) [inline src/js/components/DataFilter/DataFilter.js:136]: Where does "20" come from? What is the rationalization behind that value?

Comment (taysea) [reply src/js/components/DataFilter/DataFilter.js:136]: I think it was just a value that Eric chose as a consistent number of steps for a range. Not sure if there was any further rationalization beyond that.

Comment (halocline) [reply src/js/components/DataFilter/DataFilter.js:136]: can we do something like `const defaultSteps = 20;` to use a variable to be self documenting?

**Review by halocline — COMMENTED** [review]
small comment related to implementation documentation

**Review by halocline — APPROVED** [review]

**Review by jcfilben — APPROVED** [review]

**Pull request merged**